File: backend/api/schemas/company_settings.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class CompanySettingsCreate(BaseModel):
    """Request schema for creating company settings"""
    company_name: str
    company_mobile: Optional[str] = None
    city: Optional[str] = None
    area: Optional[str] = None
    complete_address: Optional[str] = None
    trn: Optional[str] = None
    currency: str = "AED - UAE Dirham (د.إ)"
    timezone: str = "Dubai (UAE)"
    
    # Billing Configuration
    trial_date_enabled: bool = True
    trial_date_days: int = 3
    delivery_date_enabled: bool = True
    delivery_date_days: int = 3
    advance_payment_enabled: bool = True

class CompanySettingsUpdate(BaseModel):
    """Request schema for updating company settings"""
    company_name: Optional[str] = None
    company_mobile: Optional[str] = None
    city: Optional[str] = None
    area: Optional[str] = None
    complete_address: Optional[str] = None
    trn: Optional[str] = None
    currency: Optional[str] = None
    timezone: Optional[str] = None
    
    # Billing Configuration
    trial_date_enabled: Optional[bool] = None
    trial_date_days: Optional[int] = None
    delivery_date_enabled: Optional[bool] = None
    delivery_date_days: Optional[int] = None
    advance_payment_enabled: Optional[bool] = None

class CompanySettingsResponse(BaseModel):
    """Response schema for company settings"""
    id: int
    organization_id: int
    
    # Company Information
    company_name: str
    company_mobile: Optional[str]
    city: Optional[str]
    area: Optional[str]
    complete_address: Optional[str]
    trn: Optional[str]
    currency: str
    timezone: str
    
    # Billing Configuration
    trial_date_enabled: bool
    trial_date_days: int
    delivery_date_enabled: bool
    delivery_date_days: int
    advance_payment_enabled: bool
    
    # Metadata
    created_at: datetime
    updated_at: datetime
    created_by: int
    
    class Config:
        from_attributes = True

File: frontend/src/services/companySettings.ts
import { apiRequest } from '../utils/api';

export interface CompanySettings {
  id: number;
  organization_id: number;
  company_name: string;
  company_mobile?: string;
  city?: string;
  area?: string;
  complete_address?: string;
  trn?: string;
  currency: string;
  timezone: string;
  trial_date_enabled: boolean;
  trial_date_days: number;
  delivery_date_enabled: boolean;
  delivery_date_days: number;
  advance_payment_enabled: boolean;
  created_at: string;
  updated_at: string;
  created_by: number;
}

export interface CompanySettingsUpdate {
  company_name?: string;
  company_mobile?: string;
  city?: string;
  area?: string;
  complete_address?: string;
  trn?: string;
  currency?: string;
  timezone?: string;
  trial_date_enabled?: boolean;
  trial_date_days?: number;
  delivery_date_enabled?: boolean;
  delivery_date_days?: number;
  advance_payment_enabled?: boolean;
}

export interface CompanySettingsCreate {
  company_name: string;
  company_mobile?: string;
  city?: string;
  area?: string;
  complete_address?: string;
  trn?: string;
  currency: string;
  timezone: string;
  trial_date_enabled: boolean;
  trial_date_days: number;
  delivery_date_enabled: boolean;
  delivery_date_days: number;
  advance_payment_enabled: boolean;
}

export const companySettingsService = {
  /**
   * Get company settings for the current organization
   */
  async getSettings(): Promise<CompanySettings> {
    return apiRequest('/api/company-settings') as Promise<CompanySettings>;
  },

  /**
   * Update company settings (creates if doesn't exist)
   */
  async updateSettings(settings: CompanySettingsUpdate): Promise<CompanySettings> {
    return apiRequest('/api/company-settings', {
      method: 'PUT',
      body: JSON.stringify(settings),
      headers: {
        'Content-Type': 'application/json',
      },
    }) as Promise<CompanySettings>;
  },

  /**
   * Create new company settings
   */
  async createSettings(settings: CompanySettingsCreate): Promise<CompanySettings> {
    return apiRequest('/api/company-settings', {
      method: 'POST',
      body: JSON.stringify(settings),
      headers: {
        'Content-Type': 'application/json',
      },
    }) as Promise<CompanySettings>;
  },
};
